src/dmd/dmodule.d

```diff
@@ -535,6 +501,77 @@ extern (C++) final class Module : Package
         return m;
     }
 
+    /**
+     * Converts a chain of identifiers to the filename of the module
+     *
+     * Params:
+     *  packages = the names of the "parent" packages
+     *  ident = the name of the child package or module
+     *
+     * Returns:
+     *  the filename of the child package or module
+     */
+    static const(char)* getFilename(Identifiers* packages, Identifier ident)
```

Comment:
Here we're returning a pointer to a string, yet what we do with it is call `toDString()` on the result. Have it return a D string in the first place.

Reply:
That was the initial version I had, but some of the CI machines were choking on it. Something about not being able to map `const(char)[]` to C++.

Reply:
Provide another function that converts the D string to a pointer for the C++ code. The DMD front end should center around using D strings - pointers should be only for the C++ code.